lib/puppet/provider/exec/powershell.rb

```diff
@@ -25,17 +28,69 @@
         }
   EOT
 
+  def self.upgrade_message
+    Puppet.warning <<-UPGRADE
+The current Puppet version is outdated and uses a library that was
+previously necessary on the current Ruby verison to support a colored console.
+
+Unfortunately this library prevents the PowerShell module from using a shared
+PowerShell process to dramatically improve the performance of resource
+application.
+
+To enable these improvements, it is suggested to upgrade to any x64 version of
+Puppet (including 3.x), or to a Puppet version newer than 3.x.
```

Comment:
This states `3.x`, but `powershell_manager` is only included on `> 3.8.5` above.